cmd/snap-bootstrap/cmd_scan_disk.go

```diff
@@ -132,9 +164,21 @@ func probePartitions(node string) ([]Partition, error) {
 
 	ret := make([]Partition, 0)
 	for _, partition := range partitions.GetPartitions() {
-		label := partition.GetName()
-		uuid := partition.GetUUID()
-		ret = append(ret, Partition{label, uuid})
+		name := partition.GetName()
+		if name == "" && fallbackMode {
```

Comment:
Maybe instead of using fallbackMode, you could do one or another thing depending on result of `isGpt()`. I think that we actually do not need that variable at all.

Thinking about this, the call to `probePartitions` could happen before the call to `bootFindPartitionUUIDForBootedKernelDisk`, as this is done always both if the uuid is found or not. Furthermore, at that point we might want to call `bootFindPartitionUUIDForBootedKernelDisk` only if this is GPT. But maybe this is more of a TODO.

Reply:
Updated, thank you for this input. Please re-review